cli/azd/pkg/alphafeatures/alpha_feature_manager.go

```diff
@@ -0,0 +1,91 @@
+package alphafeatures
+
+import (
+	"log"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+)
+
+type AlphaFeatureManager struct {
+	configManager   config.UserConfigManager
+	userConfigCache config.Config
+	// used for mocking alpha features on testing
+	alphaFeaturesResolver func() []AlphaFeature
+}
+
+func NewAlphaFeaturesManager(configManager config.UserConfigManager) *AlphaFeatureManager {
+	return &AlphaFeatureManager{
+		configManager: configManager,
+	}
+}
+
+func (m *AlphaFeatureManager) ListFeatures() (map[string]AlphaFeature, error) {
+	result := make(map[string]AlphaFeature)
+
+	var alphaFeatures []AlphaFeature
+	if m.alphaFeaturesResolver != nil {
+		alphaFeatures = m.alphaFeaturesResolver()
+	} else {
+		alphaFeatures = mustUnmarshalAlphaFeatures()
+	}
+
+	for _, aFeature := range alphaFeatures {
+		// cast is safe here from string to AlphaFeatureId
+		status := disabledText
+		if m.IsEnabled(AlphaFeatureId(aFeature.Id)) {
+			status = enabledText
+		}
+
+		result[aFeature.Id] = AlphaFeature{
+			Id:          aFeature.Id,
+			Description: aFeature.Description,
+			Status:      status,
+		}
+	}
+
+	return result, nil
+}
+
+func (m *AlphaFeatureManager) IsEnabled(featureId AlphaFeatureId) bool {
+	if m.userConfigCache == nil {
```

Comment:
Use `sync.Once` to do lazy-loads like this. It enforces at-most once initialization and avoids reader/writer problems.

We've expanded our multithreadedness in the application with the introduction of `async` everywhere, and it might be good to make these singleton components play well in that world.